server/lib/pbench/indexer.py

```diff
@@ -3527,21 +3527,31 @@ def gen_toc(self):
         toc_dirs = _dict_const()
         for m in self.members:
             # Always strip the prefix
-            name = m.name[prefix_l:]
+            path = m.name[prefix_l:]
             if m.isdir():
-                dname = name if name.endswith(os.path.sep) else "{}{}".format(
-                    name, os.path.sep)
-                if dname in toc_dirs:
+                if path == "/" or path == "":
+                    dpath = "/"
+                    name = None
+                    path_els = []
+                else:
+                    dpath = path[:-1] if path.endswith(os.path.sep) else path
+                    name = os.path.basename(dpath)
+                    path_els = dpath.split(os.path.sep)[1:-1]
```

Comment:
Would it make sense to have a check somewhere that `dpath` always starts with an `os.path.sep`? The split result would be strange if that were not the case.

Reply:
I just noticed a typo in the commit message: 

"measurement_units" is not called "measurement_title": not-->now

Reply:
> Would it make sense to have a check somewhere that `dpath` always starts with an `os.path.sep`? The split result would be strange if that were not the case.

Not sure that is necessary.  Before we get to this point of processing the TOC entries, the `PbenchTarBall` constructor at line 3650 has already verified that all members have the proper prefix.  With that, when we strip the `prefix_l` length off the member's `name`, we should always have a leading `os.path.sep`, except in the case of the top-level directory, which we handle at line 3532.